gauged/drivers/mysql.py

```diff
@@ -196,6 +198,34 @@ def clear_from(self, offset, timestamp):
         execute('''UPDATE gauged_writer_history SET timestamp = %s
             WHERE timestamp > %s''', (timestamp, timestamp))
 
+    def clear_key_before(self, key, namespace, timestamp=None):
+        namespace_key = (namespace, key)
+        translated_key = self.lookup_ids((namespace_key,)).get(namespace_key)
+        execute = self.cursor.execute
+        if timestamp is not None:
+            params = (translated_key, namespace, timestamp)
+            execute('''DELETE FROM gauged_data WHERE `key` = %s AND namespace = %s AND offset <= %s''', params)
```

Comment:
The only remaining issue I can see is that you're passing in a `timestamp` here but the table stores block offsets which are equal to `timestamp // config.block_size`.

You'll need to calculate the block offset in the writer (where the config is available) and then pass the `offset` and `timestamp` to the driver methods. Have a look at [`writer#clear_from`](https://github.com/chriso/gauged/blob/37198211702973cbf0aafe08594fc0244d067068/gauged/writer.py#L169-176) and [`driver#clear_from`](https://github.com/chriso/gauged/blob/37198211702973cbf0aafe08594fc0244d067068/gauged/drivers/sqlite.py#L183-190) for an example.